validations/src/validations.rs

```diff
@@ -264,13 +264,26 @@ pub fn validate_reveal_transaction(
 
     // Get DataRequest information
     let dr_pointer = re_tx.body.dr_pointer;
-    let dr_output = dr_pool
-        .get_dr_output(&dr_pointer)
+    let dr_state = dr_pool
+        .data_request_pool
+        .get(&dr_pointer)
         .ok_or(TransactionError::DataRequestNotFound { hash: dr_pointer })?;
 
-    // TODO: Validate commitment
+    let pkh = re_tx.signatures[0].public_key.pkh();
```

Comment:
You could make `validate_commit_reveal_signature` function to return a reference to the first signature and you would avoid the unsafe `signatures[0]` access here.